web/source/osk/defaultLayouts.ts

```diff
@@ -238,16 +247,17 @@ namespace com.keyman.osk {
               // Fall back to US English keycap text as default for the base two layers if not otherwise defined.
               // (Any 'ghost' keys must be explicitly defined in layout for these layers.)
               if(isDefault) {
-                if((key['text'] == '' || typeof key['text'] == 'undefined') &&  key['id'] != 'K_SPACE' && kx+65 * isShift < Layouts.dfltText.length) {
-                  key['text']=Layouts.dfltText[kx+65*isShift];
+                if(key['id'] != 'K_SPACE' && kx+65 * isShift < Layouts.dfltText.length && key['text'] !== null) {
```

Comment:
This is a semantic change. Can you explain why you made the change. Not disputing that the change is required, just want to make sure I understand your reason for making the change :)

Reply:
Sure.

First, note the change to the assignment line:  `key['text'] = key['text'] || /* original assignment */`  If the key already has assigned text, it simply keeps the existing text; in effect, this section performs the same check as the old code's `(key['text'] == '' || typeof key['text'] == 'undefined')` portion of the if-condition.

The one distinct change here is that the if-condition now checks for `key['text'] !== null` - this section would _not_ have passed the former condition's check, since `typeof null != 'undefined'` and `null != ''`.  

In effect, I made sure to keep the actual semantics the same, though the phrasing and expression are different.  We've always been filtering on `key['text'] !== null`, though that may not have been obvious... or even intended.  This just makes it much more obvious that we're doing it, and thus makes the code a bit clearer.